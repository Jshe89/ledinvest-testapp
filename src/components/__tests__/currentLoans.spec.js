import currentLoans from '../../mocks/currentLoans.json';

test('the pizza data is correct', () => {
  expect(currentLoans).toMatchSnapshot();
  expect(currentLoans.loans).toHaveLength(3);
});

for (let i = 0; i < currentLoans.loans.length; i += 1) {
  it(`currentLoans[${i}] should have properties (id,
    title,
    tranche,
    available,
    annualised_return,
    term_remaining,
    ltv,
    amount)`, () => {
    expect(currentLoans.loans[i]).toHaveProperty('id');
    expect(currentLoans.loans[i]).toHaveProperty('title');
    expect(currentLoans.loans[i]).toHaveProperty('tranche');
    expect(currentLoans.loans[i]).toHaveProperty('available');
    expect(currentLoans.loans[i]).toHaveProperty('annualised_return');
    expect(currentLoans.loans[i]).toHaveProperty('term_remaining');
    expect(currentLoans.loans[i]).toHaveProperty('ltv');
    expect(currentLoans.loans[i]).toHaveProperty('amount');
  });
}
