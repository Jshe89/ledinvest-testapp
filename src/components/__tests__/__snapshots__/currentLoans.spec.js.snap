// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the pizza data is correct 1`] = `
Object {
  "loans": Array [
    Object {
      "amount": "85,754",
      "annualised_return": "8.60",
      "available": "11,959",
      "id": "1",
      "ltv": "48.80",
      "term_remaining": "864000",
      "title": "Voluptate et sed tempora qui quisquam.",
      "tranche": "A",
    },
    Object {
      "amount": "85,754",
      "annualised_return": "7.10",
      "available": "31,405",
      "id": "5",
      "ltv": "48.80",
      "term_remaining": "1620000",
      "title": "Consectetur ipsam qui magnam minus dolore ut fugit.",
      "tranche": "B",
    },
    Object {
      "amount": "85,754",
      "annualised_return": "4.80",
      "available": "12,359",
      "id": "12",
      "ltv": "48.80",
      "term_remaining": "879000",
      "title": "Dolores repudiandae ut voluptas unde laborum quaerat et sapiente.",
      "tranche": "C",
    },
  ],
}
`;
